libs/variant/tests/unit/serialization.cpp

```diff
@@ -108,3 +108,124 @@ TEST(VariantSerialization, ArrayArraySerialisation)
   EXPECT_EQ(y[3][2], 9);
   EXPECT_EQ(y[3][3], 10);
 }
+
+TEST(VariantSerialization, VariantInt)
+{
+  MsgPackSerializer stream;
+  Variant           i1, i2;
+
+  i1 = 123456;
+  stream << i1;
+  stream.seek(0);
+  stream >> i2;
+  EXPECT_EQ(i2.As<int>(), 123456);
+}
+
+TEST(VariantSerialization, VariantFloat)
+{
+  MsgPackSerializer stream;
+  Variant           fp1, fp2;
+
+  fp1 = 1.25;  // should be repn exactly.
+  stream << fp1;
+  stream.seek(0);
+  stream >> fp2;
+  EXPECT_EQ(fp2.As<float>(), 1.25);
+}
+
+TEST(VariantSerialization, VariantString)
+{
+  MsgPackSerializer stream;
+  Variant           s1, s2;
+
+  s1 = "123456";
+  stream << s1;
+  stream.seek(0);
+  stream >> s2;
+  EXPECT_EQ(s2.As<std::string>(), "123456");
+}
+
+TEST(VariantSerialization, VariantNull)
+{
+  MsgPackSerializer stream;
+  Variant           null1, null2;
+
+  null1 = Variant::Null();
+  stream << null1;
+  stream.seek(0);
+  stream >> null2;
+  EXPECT_EQ(null2.IsNull(), true);
+}
+
+TEST(VariantSerialization, DISABLED_VariantArray)
```

Comment:
I think this one can be re-enabled as Null should now work?

Reply:
These tests do not work. I think there is significant overlap here with the work that @engineerkatie is doing. I will raise an issue to fix these issues but I think it is out of the scope of this PR at this time

Reply:
:+1: